File: Core/Models/Branch/BranchSocialLinks.swift
import Foundation

public struct BranchSocialLinks: Codable, Identifiable, Hashable, Sendable {
    public let id: EntityID
    public var branchID: EntityID
    public var whatsappParentsLink: String?
    public var whatsappAthletesLink: String?
    public var telegramChannelLink: String?
    public var instagramHandle: String?
    public var tiktokHandle: String?
    public var youtubeChannelURL: String?
    public var websiteURL: String?

    public init(
        id: EntityID = UUID(),
        branchID: EntityID,
        whatsappParentsLink: String? = nil,
        whatsappAthletesLink: String? = nil,
        telegramChannelLink: String? = nil,
        instagramHandle: String? = nil,
        tiktokHandle: String? = nil,
        youtubeChannelURL: String? = nil,
        websiteURL: String? = nil
    ) {
        self.id = id
        self.branchID = branchID
        self.whatsappParentsLink = whatsappParentsLink
        self.whatsappAthletesLink = whatsappAthletesLink
        self.telegramChannelLink = telegramChannelLink
        self.instagramHandle = instagramHandle
        self.tiktokHandle = tiktokHandle
        self.youtubeChannelURL = youtubeChannelURL
        self.websiteURL = websiteURL
    }
}

File: flutter_app/lib/core/models/branch_social_links.dart
import 'entity_id.dart';

/// Port of Core/Models/Branch/BranchSocialLinks.swift.

class BranchSocialLinks {
  final EntityID id;
  final EntityID branchId;
  final String? whatsappParentsLink;
  final String? whatsappAthletesLink;
  final String? telegramChannelLink;
  final String? instagramHandle;
  final String? tiktokHandle;
  final String? youtubeChannelUrl;
  final String? websiteUrl;

  const BranchSocialLinks({
    required this.id,
    required this.branchId,
    this.whatsappParentsLink,
    this.whatsappAthletesLink,
    this.telegramChannelLink,
    this.instagramHandle,
    this.tiktokHandle,
    this.youtubeChannelUrl,
    this.websiteUrl,
  });

  factory BranchSocialLinks.fromJson(Map<String, dynamic> json) =>
      BranchSocialLinks(
        id: json['id'] as String,
        branchId: json['branchID'] as String,
        whatsappParentsLink: json['whatsappParentsLink'] as String?,
        whatsappAthletesLink: json['whatsappAthletesLink'] as String?,
        telegramChannelLink: json['telegramChannelLink'] as String?,
        instagramHandle: json['instagramHandle'] as String?,
        tiktokHandle: json['tiktokHandle'] as String?,
        youtubeChannelUrl: json['youtubeChannelURL'] as String?,
        websiteUrl: json['websiteURL'] as String?,
      );

  Map<String, dynamic> toJson() => {
    'id': id,
    'branchID': branchId,
    'whatsappParentsLink': whatsappParentsLink,
    'whatsappAthletesLink': whatsappAthletesLink,
    'telegramChannelLink': telegramChannelLink,
    'instagramHandle': instagramHandle,
    'tiktokHandle': tiktokHandle,
    'youtubeChannelURL': youtubeChannelUrl,
    'websiteURL': websiteUrl,
  };
}
